quantities/quantities_body.hpp

```diff
@@ -298,13 +292,45 @@ constexpr bool Quantity<D>::operator!=(Quantity const& right) const {
 template<typename D>
 void Quantity<D>::WriteToMessage(
     not_null<serialization::Quantity*> const message) const {
+  static_assert(D::length_is_serializable,
+                "Invalid length exponent");
+  static_assert(D::mass_is_serializable,
+                "Invalid mass exponent");
+  static_assert(D::time_is_serializable,
+                "Invalid time exponent");
+  static_assert(D::current_is_serializable,
+                "Invalid current exponent");
+  static_assert(D::temperature_is_serializable,
+                "Invalid temperature exponent");
+  static_assert(D::amount_is_serializable,
+                "Invalid amount exponent");
+  static_assert(D::luminous_intensity_is_serializable,
+                "Invalid luminous intensity exponent");
+  static_assert(D::angle_is_serializable,
+                "Invalid angle exponent");
   message->set_dimensions(D::representation);
   message->set_magnitude(magnitude_);
 }
 
 template<typename D>
 Quantity<D> Quantity<D>::ReadFromMessage(
     serialization::Quantity const& message) {
+  static_assert(D::length_is_serializable,
+                "Invalid length exponent");
```

Comment:
Should we keep this stuff in the message? it makes the messages larger, adds quite a bit of ugly code, and I have never seen it fail (we never change the dimensions of a field, we add new fields instead).

Reply:
Let's leave it that way for now.